rafthttp/stream.go

```diff
@@ -380,6 +400,18 @@ func (cr *streamReader) dial() (io.ReadCloser, error) {
 		return nil, err
 	case http.StatusOK:
 		return resp.Body, nil
+	case http.StatusNotFound:
+		b, err := ioutil.ReadAll(resp.Body)
```

Comment:
2.1 should never return not found for a known path. it should return bad request.